osu.Game/Screens/Play/Player.cs

```diff
@@ -657,6 +656,12 @@ protected bool PerformExit(bool skipTransition = false)
                 // - the pause / fail dialog was requested but couldn't be displayed due to the type or state of this Player instance.
                 this.Exit();
             }
+            else
+            {
+                // May be restarting from results screen.
+                if (this.GetChildScreen() != null)
+                    this.MakeCurrent();
```

Comment:
I have ignored adding back the `!ValidForResume` assertion as I think it's a given since the screen is not current and the current screen is presumed to be the results screen.